cmd/saga/internal/auth/auth.go

```diff
@@ -0,0 +1,139 @@
+package auth
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"github.com/adrg/xdg"
+	authv1beta1 "github.com/einride/saga/cmd/saga/gen/einride/saga/extend/auth/v1beta1"
+	"github.com/spf13/cobra"
+	"go.einride.tech/aip-cli/aipcli"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const (
+	authConfigFile = "saga/auth.json"
+)
+
+func Authenticate(ctx context.Context, host string, verboseLogging bool) (string, error) {
+	authFile, err := readCredentials()
+	if err != nil {
+		return "", err
+	}
+	clientID := authFile.ClientID
+	clientSecret := authFile.ClientSecret
+
+	if verboseLogging {
+		fmt.Printf(">> using client-id: %s, and client-secret with length: %d \n", clientID, len(clientSecret))
+	}
+
+	authClient, cleanup, err := authServiceClient(ctx, host)
+	if err != nil {
+		return "", err
+	}
+	defer cleanup()
+
+	request := authv1beta1.ExchangeSecretRequest{
+		ClientId:     clientID,
+		ClientSecret: clientSecret,
+	}
+
+	response, err := authClient.ExchangeSecret(ctx, &request)
+	if err != nil {
+		return "", err
+	}
+	return response.AccessToken, nil
+}
+
+type SagaAuthFile struct {
+	ClientID     string `json:"clientID"`
+	ClientSecret string `json:"clientSecret"`
+}
+
+func readCredentials() (*SagaAuthFile, error) {
+	authFilepath, err := xdg.ConfigFile(authConfigFile)
+	if err != nil {
+		return nil, err
+	}
+	_, err = os.Stat(authFilepath)
+	if err != nil {
```

Comment:
Style consistency nitpick 🤓 
```suggestion
	if _, err := os.Stat(authFilepath); err != nil {
```

Reply:
:pray: